### PR TITLE
Issue/9815 domain registration analytics

### LINKED ISSUE #9815: Use Domain Credit
All WordPress.com plans include a custom domain name. This is given to the users as a domain credit, which they can redeem against any valid domain.

This issue is to implement a way for users to redeem their domain credit in the Android app. The flow is included below.

Most of this flow is already implemented via this issue (#9739). The goal here is to expose it on the **My Site** screen. This flow will be hidden behind the DOMAIN_REGISTRATION_ENABLED feature flag.

**Tasks**

- [ ] Create a *Call to Action* on **My Site** screen (#9839)
- [ ] Add reminder message to verify email address when domain registration is completed (#9866)
- [ ] Add descriptive introductory paragraph to **Domain Suggestions** screen (#9865)
- [ ] Add analytics - Would be great to track if people are using this flow and if they are dropping off at any point

DomainCreditFlow

### PR DESCRIPTION
**Pull request by khaykov**
Fixes #9815

This PR adds tracking events for the domain registration process.

To test:

- Install the wasabi build.
- Make sure the domain registration related events are fired (or will probably fire in case of error).

Update release notes:

- [ ] If there are user facing changes, I have added an item to `RELEASE-NOTES.txt`.

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/viewmodel/domains/DomainRegistrationDetailsViewModel.kt [hunk 1]
```diff
@@ -213,6 +215,7 @@ class DomainRegistrationDetailsViewModel @Inject constructor(
     @Subscribe(threadMode = ThreadMode.MAIN)
     fun onCartRedeemed(event: OnShoppingCartRedeemed) {
         if (event.isError) {
+            AnalyticsTracker.track(Stat.AUTOMATED_TRANSFER_CUSTOM_DOMAIN_PURCHASE_FAILED)
```
Comment (planarvoid) [inline WordPress/src/main/java/org/wordpress/android/viewmodel/domains/DomainRegistrationDetailsViewModel.kt]: There is an injectable `AnalyticsTrackerWrapper` class which I think should be used in testable classes like `ViewModels`. This way you can add a check into your unit test that the event actually gets fired. It's ok to use the static methods in Fragments and Activities 👍

**Comment by khaykov** [comment]
Thanks for the suggestion! Updated VM to use AnalyticsTrackerWrapper 👍

**Comment by khaykov** [comment]
I just realized that my push failed 🤦‍♂

**Comment by khaykov** [comment]
@planarvoid Ready for another round 👍

**Pull request merged**